### DISCUSSION TITLE
workflow_dispatch picking the branch

### DISCUSSION THREAD

**Discussion by MrAntix** [community/community#18648] [Actions]
We have package publishing behind a workflow_dispatch, we have one main branch which will produce a full release and other branches which produce a pre-release versions.

1. Can we set the default for the branch picker and filter or provide the items in the list?

Currently the user hits the trigger and the default is the default branch (main) is preselected, this is a risk, the user may trigger a full release accidentally.
We can stop this by making the workflow only show branches that are not "main" or a specific branch

2.  Could we have the ability to select the branch for a workflow in the workflow its self rather than the user selecting it?

We would like to have a workflow that only runs against a predefined branch, I know we can guard against this, but having a single choice would mean the user does not have to do the branch selection at all

```
// show "next" and "alpha" as choices, "next" default
on: workflow_dispatch
    branch: [next, alpha] 
```

```
// show "next" and "alpha" as choices, no default
on: workflow_dispatch
    branch: [,next, alpha] 
```

```
// show any branch which is not "main"
on: workflow_dispatch
    branch: [!main]
```

```
// don't show picker, always run on "alpha"
on: workflow_dispatch
    branch: alpha
```

Many thanks, Ant

**Comment by MathiasReker** [comment]
Hello

I suggest using tags instead. Have a look at this example: https://github.com/MathiasReker/CI-CD/blob/develop/.github/workflows/ci-cd.yml

**Reply by MrAntix** [reply]
we're also using semantic-release, which creates the tags and calls the publish

I'll take a look in any case, thank you for the suggestion
